pkg/client/fake/client_test.go

```diff
@@ -1476,6 +1476,84 @@ var _ = Describe("Fake client", func() {
 		objOriginal.Status.NodeInfo.MachineID = "machine-id-from-status-update"
 		Expect(cmp.Diff(objOriginal, actual)).To(BeEmpty())
 	})
+
+	It("should be able to update an object after updating an object's status", func() {
+		obj := &corev1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node",
+			},
+			Spec: corev1.NodeSpec{
+				PodCIDR: "old-cidr",
+			},
+			Status: corev1.NodeStatus{
+				NodeInfo: corev1.NodeSystemInfo{
+					MachineID: "machine-id",
+				},
+			},
+		}
+		cl := NewClientBuilder().WithStatusSubresource(obj).WithObjects(obj).Build()
+		expectedObj := obj.DeepCopy()
+
+		obj.Status.NodeInfo.MachineID = "machine-id-from-status-update"
+		Expect(cl.Status().Update(context.Background(), obj)).NotTo(HaveOccurred())
+
+		obj.Annotations = map[string]string{
+			"some-annotation-key": "some",
+		}
+		expectedObj.Annotations = map[string]string{
+			"some-annotation-key": "some",
+		}
+		Expect(cl.Update(context.Background(), obj)).NotTo(HaveOccurred())
```

Comment:
This test fails. The test below was double checking that the reverse order also works.